tests/client/test_client_auth.py

```diff
@@ -434,6 +423,33 @@ def test_get_auth_token_doesnt_refreshe_if_refresh_token_and_future_expiration(
         assert client.get_auth_token() == "access"
         refresh_token.assert_not_called()
 
+    def test_client_clears_active_tenant_if_login_fails_on_initialization(
+        self, patch_post
+    ):
+        post = patch_post(
+            {
+                "errors": [
+                    {
+                        "message": "",
+                        "locations": [],
+                        "path": ["tenant"],
+                        "extensions": {"code": "UNAUTHENTICATED"},
+                    }
+                ]
+            }
+        )
+
+        # create a client just so we can use its settings methods to store settings
+        client = Client()
+        settings = client._load_local_settings()
+        settings.update(api_token="API_TOKEN", active_tenant_id=str(uuid.uuid4()))
+        client._save_local_settings(settings)
+
+        # this initialization will fail with the patched error
+        client = Client()
+        settings = client._load_local_settings()
+        assert "active_tenant_id" not in settings
```

Comment:
Will this test delete tokens from my machine?

Reply:
It should not because of this fixture: https://github.com/PrefectHQ/prefect/blob/master/tests/conftest.py#L15

This makes the home dir a temporary directory during tests

Reply:
Excellent